primefield/src/fiat.rs

```diff
@@ -128,10 +128,15 @@ macro_rules! fiat_field_arithmetic {
             ///
             /// Does not check that self is non-zero.
             const fn invert_unchecked(&self) -> Self {
+                const BITS: usize = {
+                    let mut f = [0; <$uint>::LIMBS + 1];
+                    $msat(&mut f);
+                    $crate::bigint::Uint::from_words(f).bits_vartime() as usize
+                };
                 let words = $crate::fiat_bernstein_yang_invert!(
                     &$mont_type(self.0.to_words()),
                     &$mont_type(Self::ONE.0.to_words()),
-                    size_of::<$bytes>() * 8,
+                    BITS,
```

Comment:
...so this could be:

```suggestion
                    <$fe as $crate::ff::PrimeField>::NUM_BITS
```